pkg/controller/petset/pet.go

```diff
@@ -311,3 +317,7 @@ func (d *defaultPetHealthChecker) isHealthy(pod *api.Pod) bool {
 func (d *defaultPetHealthChecker) isDying(pod *api.Pod) bool {
 	return pod != nil && pod.DeletionTimestamp != nil
 }
+
+func isEvicted(pod *api.Pod) bool {
+	return pod != nil && pod.Status.Phase == api.PodFailed && pod.Status.Reason == "Evicted"
```

Comment:
Lets just do this when phase==failed, @smarterclayton wdyt? 
I think the only situation that happens is when the kubelet rejects the pod because of admission check failure, since we disallow `restartPolicy=Never` on petset.